go/vt/sqlparser/ast_format.go

```diff
@@ -74,6 +74,10 @@ func (node *CommentOnly) Format(buf *TrackedBuffer) {
 
 // Format formats the node.
 func (node *Union) Format(buf *TrackedBuffer) {
+	if node.With != nil {
+		buf.astPrintf(node, "%v", node.With)
```

Comment:
@maxenglander Can you also add a unit test for this to the `sqlparser` package?

Reply:
`go/vt/sqlparser/parse_test.go` is a good place for this